cores/esp8266/core_esp8266_wiring_digital.cpp

```diff
@@ -147,13 +147,16 @@ void ICACHE_RAM_ATTR interrupt_handler(void *arg) {
       // to make ISR compatible to Arduino AVR model where interrupts are disabled
       // we disable them before we call the client ISR
       uint32_t savedPS = xt_rsil(15); // stop other interrupts
-      ArgStructure* localArg = (ArgStructure*)handler->arg;
-      if (localArg && localArg->interruptInfo)
-      {
-         localArg->interruptInfo->pin = i;
-         localArg->interruptInfo->value = __digitalRead(i);
-         localArg->interruptInfo->micro = micros();
-      }
+	  if (handler->functional)
+	  {
+    	  ArgStructure* localArg = (ArgStructure*)handler->arg;
+    	  if (localArg && localArg->interruptInfo)
```

Comment:
In this spot, discerning only null from non-null causes undefined behavior due to accessing arg->interruptInfo

Reply:
Right, I missed that, thanks for the explanation